examples/example_hierarchical_abstraction.jl

```diff
@@ -0,0 +1,197 @@
+using StaticArrays, JuMP, Plots, Test
```

Comment:
Seems like a lot of code duplication..

Reply:
yeah, again, there's the one that's used for documentation, and the one that's used for testing... I'm not sure what's the best way to proceed ...

Reply:
Maybe writing an example and using `include`?

Reply:
Yes, but the problem is that for tests, for example, you don't want to plot the figures. So you have to put the plots in an if() condition that must be executed if you include it in the doc but not in the tests. And a priori, you don't want this if() to appear in the doc... (a bit like the examples/example_hierarchical_abstraction.jl)..